lib/rules/no-constant-condition.js

```diff
@@ -35,9 +35,11 @@ module.exports = function(context) {
                 return isConstant(node.argument);
 
             case "BinaryExpression":
-            case "LogicalExpression":
                 return isConstant(node.left) && isConstant(node.right) && node.operator !== "in";
 
+            case "LogicalExpression":
+                return isConstant(node.left) || isConstant(node.right);
```

Comment:
I think this is a bit too broad. For instance, this isn't a constant condition:

```js
if (foo && true) {}
```

Because `foo` can be anything, this is equivalent to just `if (foo) {}`, which is also not constant.

So I think this has to be more specific to the `false && something` case rather than being general to all logical expressions.